components/encoder/AMS/ams_as5048.go

```diff
@@ -120,15 +123,16 @@ func (cfg *I2CAttrConfig) ValidateI2C(path string) error {
 	return nil
 }
 
-// AS5048 is a struct representing an instance of a hardware unit
+// Encoder is a struct representing an instance of a hardware unit
```

Comment:
in the future, you don't have to make names consistent. totally okay here - just not an expectation.